projects/miopen/src/solver/conv/conv_hip_implicit_gemm_3d_grouped_fwd_xdlops.cpp

```diff
@@ -457,27 +399,197 @@ bool ConvHipImplicitGemm3DGroupFwdXdlops::CheckCKApplicability(
+                    -> std::vector<std::string> {
+                    return miopen::solver::FillValidKernelsIDs<DeviceOpGFwdDefaultPtrs<T>,
+                                                               CKArgs<T>>(problem);
+                };
+                return miopen::solver::conv::RunParameterPredictionModel<T>(ctx,
+                                                                            problem,
+                                                                            valid_kernels,
+                                                                            index,
+                                                                            split_k,
+                                                                            kernel_id,
+                                                                            fill_valid_kernels,
+                                                                            solver_name);
+            };
+            switch(problem.GetInDataType())
+            {
+            case miopenHalf: ai_success = run_ai_heuristics(ck::half_t{}); break;
+            case miopenFloat: ai_success = run_ai_heuristics(float{}); break;
+            case miopenBFloat16: ai_success = run_ai_heuristics(ck::bhalf_t{}); break;
+            default: break;
+            }
+            if(ai_success)
+            {
+                MIOPEN_LOG_I("AI heuristics successfully selected kernel: " << kernel_id);
+                return;
+            }
+            else
+            {
+                MIOPEN_LOG_I("AI heuristics failed, falling back to fallback heuristics or default "
+                             "initialization");
+                // Continue to fallback heuristics
+            }
+        }
+    }
+    else
+    {
+
+        MIOPEN_LOG_I2("AI heuristics only supported for FP32 data type, falling back to fallback "
+                      "heuristics or default initialization");
+    }
+#endif
+
+    // 3. Fallback heuristics for BF16/FP16 only
+    if(problem.GetInDataType() == miopenBFloat16 || problem.GetInDataType() == miopenHalf)
+    {
+        switch(problem.GetInDataType())
+        {
+        case miopenHalf: Init<ck::half_t>(problem); break;
+        case miopenBFloat16: Init<ck::bhalf_t>(problem); break;
+        default: break;
+        }
+
+        auto find_kernel = [&valid_kernels = std::as_const(valid_kernels)](
+                               const std::size_t& index,
+                               const std::string& kernel_id) -> std::size_t {
+            if(index < valid_kernels.size() && valid_kernels[index] == kernel_id)
+                return index;
+            auto it = std::find(valid_kernels.begin(), valid_kernels.end(), kernel_id);
+            if(it != valid_kernels.end())
+                return static_cast<std::size_t>(it - valid_kernels.begin());
+            MIOPEN_LOG_E("Not found :" << index << "-" << kernel_id);
```

Comment:
This error message has been shown to be a bit confusing, see e.g. [SWDEV-551425](https://ontrack-internal.amd.com/browse/SWDEV-551425?focusedId=20342972&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-20342972).
While it is true that it means this hard-code heuristics has failed... this could be expected behaviour as in that ticket where the focus is on gfx950
This current PR touches the Heuristics for this solver (even though in the end we did not implement the heuristics model for BF16/FP16 in this case). So perhaps this would be a good opportunity to alter this message at the same time?

We could make it a bit more verbose, and change it to e.g. a log level 2
```suggestion
            MIOPEN_LOG_I2("BF16/FP16 heuristics did not find: " << kernel_id  << " at index: " index "; Falling back to index 0");
```

Reply:
I thought this was only going to check for these on a specific ASIC, but if we keep it as is, then I agree we shouldn't be logging error for this case.